src/Uno.UWP/Storage/ApplicationData/ApplicationDataCompositeValue.cs

```diff
@@ -0,0 +1,210 @@
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Windows.Foundation.Collections;
+
+namespace Windows.Storage;
+
+/// <summary>
+/// Represents related app settings that must be serialized and deserialized atomically.
+/// </summary>
+public partial class ApplicationDataCompositeValue :
```

Comment:
It may be worth checking if this type is thread-safe in WinUI. I don't know for sure.

Reply:
Unfortunately, sources for this one are not public. However, the composite value is only serialized and deserialized when stored in `ApplicationData` - so even if two threads retrieved it at the same time, they get separate instances, and the last one to save it wins. Because we are not handling concurrency on `ApplicationData` settings level, I guess this is fine for now.